.travis.yml

```diff
@@ -1,13 +1,15 @@
 language: cpp
-
-os: linux
+os:
+ - linux
+ - osx
 
 dist: bionic
 
 compiler:
     - gcc
     - clang
-    
+    - xcode
```

Comment:
AFAIK xcode is not a separate compiler, but xcode == clang.
meaning that it's enough to add `os: osx`, and it'll compile.

Reply:
yeah, it's weird in the matrix. Listing `xcode` here does add a Linux+XCode build which passes. So I'm not sure if it fully matches `clang` or not. The builds on OS X do fail though, which I'm more interested in trying to get fixed.

Reply:
It seems that there is a conflict
 between python2 and python3

Reply:
On the same topic, looking at the OSX build log:

```sh
GCC version

Apple LLVM version 9.1.0 (clang-902.0.39.2)

Target: x86_64-apple-darwin17.7.0

Thread model: posix

InstalledDir: /Applications/Xcode-9.4.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin

LLVM version

Apple LLVM version 9.1.0 (clang-902.0.39.2)

Target: x86_64-apple-darwin17.7.0

Thread model: posix

InstalledDir: /Applications/Xcode-9.4.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```

Both GCC and Clang point at Xcode, which in turn is just Clang.

Look at the Linux Xcode build log:
```sh
$ export CXX=${CXX:-g++}

$ export CXX_FOR_BUILD=${CXX_FOR_BUILD:-g++}

$ export CC=${CC:-gcc}

$ export CC_FOR_BUILD=${CC_FOR_BUILD:-gcc}

$ xcode --version

Command 'xcode' not found, did you mean:

  command 'code' from snap code (0ba0ca52)

  command 'tcode' from deb emboss

See 'snap info <snapname>' for additional versions.
```
Xcode doesn't exist there, and it just compiles with GCC.

I think we can remove Xcode from the `compiler` list.